src/iface/service-link-iface.c

```diff
@@ -144,13 +144,14 @@ static const char *_get_arg_line(const char *str, const char *key_eq, size_t *si
 		goto out;
 
 	str_end = str + strlen(str);
-	for (str_end = str + strlen(str); str < str_end; str = line_end + 1) {
-		line_end = strchr(str, '\n') ? : str_end;
-
-		if (!strcmp(key_eq, str)) {
-			/* get the value and its size */
-			str += strlen(key_eq);
-
+	for (str_end = str + strlen(str); str < str_end; str = line_end) {
+		line_end = strchr(str, '\n');
+		if (line_end)
+			line_end++;
+		else
+			return NULL;
+
+		if (!strncmp(key_eq, str, strlen(key_eq))) {
```

Comment:
I'd lean here to the way where we still allow the end of the overall arg string to be just \0, not \n\0, because it could be very easy to forget the \n at the end, mainly if it's used outside as a library function by others.

So I'd probably leave the existing loop and checks as they were (also returning just the value) and then let the _do_service_link_notify decorate the value with the proper key. This will also let us use a different key for the actual notification if needed (or different format than key=value pair). That "decoration" could all happen in _do_service_link_notify then (...or calling out specific functions centrally from there).

Of course, the strcmp is a bug, there needs to be the strncmp as you patched...